sky/clouds/cloud.py

```diff
@@ -366,6 +367,7 @@ def is_label_valid(cls, label_key: str,
         del label_key, label_value
         return True, None
 
+    @timeline.event
     def get_feasible_launchable_resources(
```

Comment:
Just noting that adding `timeline.event` decorator has a tiny* (~1ms per invocation on my mac), but non-trivial at scale impact on perf. I measured get_feasible_launchable_resources is invoked 5 times in a `sky launch` in my environment, so this should be fine to add here, but just wanted to take a note of this.

------------
*If you're curious, I measured with this script by adding/removing `@timeline.event` decorator on main :
```
from sky.utils import timeline
import time
import functools

# Add this helper decorator to measure time
def measure_time(func):
    @functools.wraps(func)
    def wrapper(*args, **kwargs):
        start_time = time.perf_counter()
        result = func(*args, **kwargs)
        end_time = time.perf_counter()
        print(f"Original function '{func.__name__}' took: {end_time - start_time:.6f} seconds")
        return result
    return wrapper

# Apply both decorators
@measure_time
@timeline.event
def main():
    print("hi")

if __name__ == "__main__":
    main()
```

Reply:
We could probably just skip recording anything if `SKYPILOT_TIMELINE_FILE_PATH` is not set - hopefully that would alleviate the penalty.